resources/migrations/0016_auto_20151028_1653.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import autoslug.fields
+import resources.models.utils
+
+
+class Migration(migrations.Migration):
```

Comment:
Yes the field needs to allow empty values at first, and then not allow them after populating the field. The problem is that you should not have schema altering changes and RunPython stuff in the same migration when using PostgreSQL according to Django docs: 

"Thus, on PostgreSQL, for example, you should avoid combining schema changes and RunPython operations in the same migration or you may hit errors like OperationalError: cannot ALTER TABLE "mytable" because it has pending trigger events."

At first I tried combining the migrations into one, but that resulted in the error. Using null=True unique=True from the beginning might work, then the migration would need to change the uniqueness, but still that would be against the recommendations.

Actually I'm not even sure about keeping the RunPython operations in the same migration as the field creations, but that seems to work without problems. 

Django docs solves this situation with three migrations [https://docs.djangoproject.com/en/1.8/howto/writing-migrations/#migrations-that-add-unique-fields]